src/bundles/ipfs-provider.js

```diff
@@ -150,14 +158,20 @@ const readSetting = (id) => {
     }
 
     try {
-      return JSON.parse(setting)
+      return JSON.parse(setting || '')
     } catch (_) {
       // res was probably a string, so pass it on.
       return setting
```

Comment:
Well return type of this function is `string | null` so `null` is fine. It maybe reasonable to change type to `string` and check for `''`, but it would be bit trickier to catch all the places where `''` case isn't considered because type checker won't really help, unlike spotting cases where `string | null` is treated as string.